onnxruntime/test/providers/cpu/tensor/transpose_test.cc

```diff
@@ -44,29 +44,19 @@ TEST(TransposeOpTest, PermRankDoesNotMatchTensorRank) {
 // Those tests will fallback to other EPs.
 
 template <class T>
-void TransposeTest(std::vector<int64_t>& input_shape,
-                   std::vector<T>& input_vals,
-                   std::vector<int64_t>* p_perm,
-                   std::vector<int64_t> expected_shape,
-                   std::initializer_list<T>& expected_vals,
-                   bool is_tensorrt_supported = true,
-                   bool is_openvino_supported = true) {
+void TransposeTest(const std::vector<int64_t>& input_shape,
+                   const std::vector<T>& input_vals,
+                   const std::vector<int64_t>* p_perm,
+                   const std::vector<int64_t>& expected_shape,
+                   const std::vector<T>& expected_vals,
+                   const std::unordered_set<std::string>& excluded_provider_types = {}) {
```

Comment:
side note - with the new BaseTester::Config* methods it might be simpler/more flexible for this sort of test structure to take a std::function<void(OpTester&)> so each test can do whatever config it needs via that.

Reply:
noted - will keep that in mind for future refactoring :)